examples/radiation_transport/radiation_calculation.py

```diff
@@ -0,0 +1,194 @@
+eq_name = "DN-DEMO_eqref.json"
+eq_name = os.sep.join([read_path, eq_name])
+eq = Equilibrium.from_eqdsk(eq_name, load_large_file=True)
+
+# %%[markdown]
+# Get first wall shape
+
+# %%
+read_path = get_bluemira_path("radiation_transport/test_data", subfolder="tests")
+fw_name = "DN_fw_shape.json"
+fw_name = os.sep.join([read_path, fw_name])
+fw_shape = Loop.from_file(fw_name)
+
+# %%[markdown]
+# Run particle solver
+
+# %%
+p_solver_params = ParameterFrame()
+solver = ChargedParticleSolver(p_solver_params, eq, dx_mp=0.001)
+solver.analyse(first_wall=fw_shape)
+
+# %%[markdown]
+# Run PROCESS solver
+
+# %%
+PROCESS_PATH = ""
+binary = f"{PROCESS_PATH}process"
+
+new_params = {
+    "kappa": 1.6969830041844367,
+}
+params = Configuration(new_params)
+
+build_config = {
+    "mode": "run",
+    "binary": binary,
+}
+
+process_solver = process.Solver(
+    params=params,
+    build_config=build_config,
+)
+process_solver.run()
+
+# %%[markdown]
+# Get impurity fractions
+
+# %%
+impurity_content = {
+    "H": process_solver.get_species_fraction("H"),
+    "He": process_solver.get_species_fraction("He"),
+    "Xe": process_solver.get_species_fraction("Xe"),
+    "W": process_solver.get_species_fraction("W"),
+}
+
+# %%[markdown]
+# Get impurity data: temperature reference
+# and radiative loss function reference
+
+# %%
+impurity_data = {
+    "H": {
+        "T_ref": process_solver.get_species_data("H")[0],
+        "L_ref": process_solver.get_species_data("H")[1],
+    },
+    "He": {
+        "T_ref": process_solver.get_species_data("He")[0],
+        "L_ref": process_solver.get_species_data("He")[1],
+    },
+    "Xe": {
+        "T_ref": process_solver.get_species_data("Xe")[0],
+        "L_ref": process_solver.get_species_data("Xe")[1],
+    },
+    "W": {
+        "T_ref": process_solver.get_species_data("W")[0],
+        "L_ref": process_solver.get_species_data("W")[1],
+    },
+}
+
+# %%[markdown]
+# Eventually customise plasma parameters to deviate from default values
+
+# %%
+plasma_params = ParameterFrame(
+    # fmt: off
+    [
+        ["kappa", "Elongation", 3, "dimensionless", None, "Input"],
+    ]
+    # fmt: on
+)
+
+# %%[markdown]
+# Eventually customise radiation parameters to deviate from default values
+
+# %%
+rad_params = ParameterFrame(
+    # fmt: off
+    [
+        ["p_sol", "power entering the SoL", 300e6, "W", None, "Input"],
+    ]
+    # fmt: on
+)
+
+# %%[markdown]
+# Run core radiation source calculation for Spherical Tokamak
+
+# %%
+stcore = STCore(solver, impurity_content, impurity_data, plasma_params, rad_params)
+
+# %%[markdown]
+# Build radiation profile at the midplane and 2D core radiation map
+
+# %%
+stcore.build_mp_rad_profile()
+stcore.build_core_radiation_map()
+
+# %%[markdown]
+# Run scrape-off layer radiation source calculation for Spherical Tokamak
+
+# %%
+stsol = STScrapeOffLayer(
+    solver, impurity_content, impurity_data, plasma_params, rad_params, fw_shape
```

Comment:
Why not calculate the values along flux surfaces rather than along this strange grid?